test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1860,10 +1859,7 @@ public static CreateIndexResponse createIndex(RestClient client, String name, Se
 
         if (settings != null && settings.getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true) == false) {
             expectSoftDeletesWarning(request, name);
-        } else if (isSyntheticSourceConfiguredInMapping(mapping)
```

Comment:
Remove logic that allows deprecation warning because of _source.mode mapping attribute usage.